controllers/routemap_test.go

```diff
@@ -214,3 +217,136 @@ func TestGetObservedGeneration(t *testing.T) {
 		assert.Equal(t, tt.ok, o)
 	}
 }
+
+func TestCleanUnstructured(t *testing.T) {
+	deployment := unstructured.Unstructured{
+		Object: map[string]interface{}{
+			"apiVersion": "apps/v1",
+			"kind":       "Deployment",
+			"metadata": map[string]interface{}{
+				"namespace": "myNamespace",
+				"name":      "myName", // remove name
+				"labels": map[string]interface{}{
+					"myLabel":        "myLabelValue",
+					weightAnnotation: "50", // remove weightAnnotation
+				},
+				"finalizers": []interface{}{ // remove finalizers
+					"finalizer.extensions/v1beta1",
+				},
+			},
+			"spec": map[string]interface{}{
+				"containers": []map[string]interface{}{
+					{
+						"command": []string{
+							"/bin/iter8",
+						},
+					},
+				},
+			},
+			"status": map[string]interface{}{ // remove status
+				"startTime": "2023-05-22T18:07:51Z",
+			},
+		},
+	}
+
+	cleaned := cleanUnstructured(&deployment)
+
+	expected := unstructured.Unstructured(
+		unstructured.Unstructured{
+			Object: map[string]interface{}{
+				"apiVersion": "apps/v1",
+				"kind":       "Deployment",
+				"metadata": map[string]interface{}{
+					"namespace": "myNamespace",
+					"labels": map[string]interface{}{
+						"myLabel": "myLabelValue",
+					},
+					"finalizers": []interface{}{},
+				},
+				"spec": map[string]interface{}{
+					"containers": []map[string]interface{}{
+						{
+							"command": []string{
+								"/bin/iter8",
+							},
+						},
+					},
+				},
+				"status": map[string]interface{}{},
+			},
+		},
+	)
+
+	assert.Equal(t, &expected, cleaned)
+}
+
+type testInformer struct {
+	o runtime.Object
+}
+
+func (i testInformer) Informer() cache.SharedIndexInformer {
+	return nil
+}
+
+func (i testInformer) Lister() cache.GenericLister {
+	return testLister(i)
+}
+
+type testLister struct {
+	o runtime.Object
+}
+
+func (l testLister) List(selector labels.Selector) (ret []runtime.Object, err error) {
+	return nil, nil
+}
+
+func (l testLister) Get(name string) (runtime.Object, error) {
+	return nil, nil
+}
+
+func (l testLister) ByNamespace(namespace string) cache.GenericNamespaceLister {
+	return testGenericLister(l)
+}
+
+type testGenericLister struct {
+	o runtime.Object
+}
+
+func (gl testGenericLister) List(selector labels.Selector) (ret []runtime.Object, err error) {
+	return nil, nil
+}
+
+func (gl testGenericLister) Get(name string) (runtime.Object, error) {
+	return gl.o, nil
+}
```

Comment:
These structs/methods are used to mock the app informer for testing `computeSignature()`